scratch/testfile.R

```diff
@@ -0,0 +1,484 @@
+  # data
+  lab_site_reporting_freq <- abs(rnorm(
+    n = n_lab_sites, mean = mean_reporting_freq,
+    sd = sd_reporting_freq
+  ))
+  lab_site_reporting_latency <- pmax(1, ceiling(rnorm(
+    n = n_lab_sites,
+    mean = mean_reporting_latency, sd = sd_reporting_latency
+  )))
+  # Set a lab-site-specific LOD in log scale
+  lod_lab_site <- rnorm(n_lab_sites, mean = mean_log_lod, sd = sd_log_lod)
+
+  # Delay distributions: Note, these are COVID specific, and we will
+  # generally want to specify these outside model configuration
+
+  # Double censored, zero truncated, generation interval
+  generation_interval <- simulate_double_censored_pmf(
+    max = gt_max, meanlog = mu_gi, sdlog = sigma_gi, fun_dist = rlnorm, n = 5e6
+  ) |> drop_first_and_renormalize()
+
+  # Set infection feedback to generation interval
+  infection_feedback_pmf <- generation_interval
+  infection_feedback_rev_pmf <- rev(infection_feedback_pmf)
+  infection_feedback <- 0
+  if_feedback <- 1
+
+  # Delay from infection to hospital admission: incubation period +
+  # time from symptom onset to hospital admission
+
+  # Get incubation period for COVID.
+  inc <- make_incubation_period_pmf(
+    backward_scale, backward_shape, r
+  )
+  sym_to_hosp <- make_hospital_onset_delay_pmf(neg_binom_mu, neg_binom_size)
+
+  # Infection to hospital admissions delay distribution
+  inf_to_hosp <- make_reporting_delay_pmf(inc, sym_to_hosp)
+
+  # Shedding kinetics delay distribution
+  vl_trajectory <- model$functions$get_vl_trajectory(
+    t_peak_mean, viral_peak_mean,
+    duration_shedding_mean, gt_max
+  )
+
+  # Generate the state level weekly R(t) before infection feedback
+  # Adds a bit of noise, can add more...
+  unadj_r_weeks <- (r_in_weeks * rnorm(length(r_in_weeks), 1, 0.03))[1:n_weeks]
+
+  # Convert to daily for input into renewal equation
+  ind_m <- get_ind_m(ot + ht, n_weeks)
+  unadj_r <- ind_m %*% unadj_r_weeks
+
+
+  # Generate the site-level expected observed concentrations -----------------
+  # first by adding variation to the site-level R(t) in each site,
+  # and then adding lab-site level multiplier and observation error
+
+
+  ### Generate the site level infection dynamics-------------------------------
+  new_i_over_n_site <- matrix(nrow = n_sites + 1, ncol = (uot + ot + ht))
+  r_site <- matrix(nrow = n_sites + 1, ncol = (ot + ht))
+  # Generate site-level R(t)
+  log_r_state_week <- log(unadj_r_weeks)
+  initial_growth_site <- vector(length = n_sites + 1)
+  log_i0_over_n_site <- vector(length = n_sites + 1)
+  for (i in 1:(n_sites + 1)) {
+    # Generate deviations in the initial growth rate and initial incidence
+    initial_growth_site[i] <- rnorm(
+      n = 1, mean = initial_growth,
+      sd = initial_growth_prior_sd
+    )
+    # This is I0/N at the first unobserved time
+    log_i0_over_n_site[i] <- rnorm(
+      n = 1, mean = log_i0_over_n,
+      sd = 0.5
+    )
+  }
+
+  log_r_site <- spatial_rt_process_rng(
```

Comment:
I am surprised you didn't need `model$functions$spatial_rt_process_rng` here!

Reply:
I used the expose functions, but next time I will use that because there was a couple of naming conflicts with stan and r functions having the same names so I renamed the r functions that were causing the problem with `_r.R`.